src/options/options_handler.h

```diff
@@ -44,72 +44,73 @@ class OptionsHandler {
+    }
   }
-
-  void unsignedLessEqual2(const std::string& option, unsigned value) {
-    options::less_equal(2)(option, value);
-  }
-
-  void doubleGreaterOrEqual0(const std::string& option, double value) {
-    options::greater_equal(0.0)(option, value);
-  }
-
-  void doubleLessOrEqual1(const std::string& option, double value) {
-    options::less_equal(1.0)(option, value);
+  template<typename T>
+  void betweenZeroAndOne(const std::string& option, const std::string& flag, T value) const {
+    if (value < 0 || value > 1) {
+      std::stringstream ss;
+      ss << flag << ": " << value << " is not a legal setting, should be 0 <= " << flag << " <= 1.";
+      throw OptionException(ss.str());
+    }
   }
 
   // theory/quantifiers/options_handlers.h
-  void checkInstWhenMode(std::string option, InstWhenMode mode);
+  void checkInstWhenMode(const std::string& option, const std::string& flag, InstWhenMode mode);
 
   // theory/bv/options_handlers.h
-  void abcEnabledBuild(std::string option, bool value);
-  void abcEnabledBuild(std::string option, std::string value);
+  void abcEnabledBuild(const std::string& option, const std::string& flag, bool value);
+  void abcEnabledBuild(const std::string& option, const std::string& flag, std::string value);
 
-  template<class T> void checkSatSolverEnabled(std::string option, T m);
+  template<class T> void checkSatSolverEnabled(const std::string& option, const std::string& flag, T m);
 
-  void checkBvSatSolver(std::string option, SatSolverMode m);
-  void checkBitblastMode(std::string option, BitblastMode m);
+  void checkBvSatSolver(const std::string& option, const std::string& flag, SatSolverMode m);
+  void checkBitblastMode(const std::string& option, const std::string& flag, BitblastMode m);
 
-  void setBitblastAig(std::string option, bool arg);
+  void setBitblastAig(const std::string& option, const std::string& flag, bool arg);
 
   // printer/options_handlers.h
-  InstFormatMode stringToInstFormatMode(std::string option, std::string optarg);
+  InstFormatMode stringToInstFormatMode(const std::string& option, const std::string& flag, std::string optarg);
 
   // decision/options_handlers.h
-  void setDecisionModeStopOnly(std::string option, DecisionMode m);
+  void setDecisionModeStopOnly(const std::string& option, const std::string& flag, DecisionMode m);
 
   /**
    * Throws a ModalException if this option is being set after final
    * initialization.
    */
-  void setProduceAssertions(std::string option, bool value);
+  void setProduceAssertions(const std::string& option, const std::string& flag, bool value);
 
-  void setStats(const std::string& option, bool value);
+  void setStats(const std::string& option, const std::string& flag, bool value);
 
-  unsigned long limitHandler(std::string option, std::string optarg);
-  void setResourceWeight(std::string option, std::string optarg);
+  unsigned long limitHandler(const std::string& option, const std::string& flag, std::string optarg);
```

Comment:
Ditto (clang-format).

Can you also change the type to uint64_t since you are already touching the line?